onnx/defs/tensor/defs.cc

```diff
@@ -1430,28 +1430,30 @@ ONNX_OPERATOR_SET_SCHEMA(
           auto scales = ctx.getInputData(1);
           if (nullptr != scales) {
             // Infer output shape's dimension value if 'scales' is known.
-            if (scales->data_type() == TensorProto::FLOAT &&
-                scales->float_data_size() == input_shape.dim_size()) {
-              for (int i = 0; i < input_shape.dim_size(); ++i) {
-                int64_t dim_value = static_cast<int64_t>(std::floor(
-                    static_cast<float>(input_shape.dim(i).dim_value()) *
-                    scales->float_data(i)));
-                if (output_shape->dim_size() > i) {
-                  if (output_shape->dim(i).has_dim_value()) {
-                    if (output_shape->dim(i).dim_value() != dim_value) {
-                      fail_shape_inference(
-                          "Dimension value inferred (",
-                          dim_value,
-                          ") is not equal to the existing dim value (",
-                          output_shape->dim(i).dim_value(),
-                          ").");
+            if (scales->data_type() == TensorProto::FLOAT){
+              const auto& data = ParseData<float>(scales);
+              if(data.size() == input_shape.dim_size()){
+                for (int i = 0; i < input_shape.dim_size(); ++i) {
+                  int64_t dim_value = static_cast<int64_t>(std::floor(
+                      static_cast<float>(input_shape.dim(i).dim_value()) *
```

Comment:
Hi @lara-hdr - can you please wait for @raymondxyang 's fix for this ? We have some other fixes like handling symbolic dims (this code doesn't have a check `has_dim_value()`) and while we are at it - we can fix `Upsample` as well which is very similar to this....

Reply:
Sounds good. Thanks.